vendor/github.com/HomesNZ/elastic/scan.go

```diff
@@ -88,6 +89,15 @@ func (s *ScanService) Fields(fields ...string) *ScanService {
 	return s
 }
 
+// Body sets the raw body to send to Elasticsearch. This can be e.g. a string,
+// a map[string]interface{} or anything that can be serialized into JSON.
+// Notice that setting the body disables the use of SearchSource and many
+// other properties of the ScanService.
+func (s *ScanService) Body(body interface{}) *ScanService {
```

Comment:
if you use function on pointer, and make changes using a pointer, why do you need to return it?